pkg/api/controllers/volume.go

```diff
@@ -105,17 +116,26 @@ func (v *VolumePortal) CreateVolume() {
 		return
 	}
 	defer v.CtrClient.Close()
+	var profId string
+	var poolID string
+	if volume.ProfileId == "" {
```

Comment:
Wrong ProfileId will not be validated. If user provided wrong profileid, it should get validated and throw error message at API level only, before code flow goes to controller.

Reply:
@PravinRanjan10 Here if condition block is when prfoilr id is not specified. incase of profileID is not cpecified the validation happens in Controller C.GetProfile() and the error reponse is like ""get profile failed: Wrong resource uuid provided!"

Reply:
As per discussed, please handle the case when poolid and wrong profileId given.

Reply:
@PravinRanjan10 , Thaks for the detailed review.
I have handled the case of having both profileid and pooid in request. also added validation ofr wrong poolid.